### PR TITLE
Updated dependencies and added tests

### LINKED ISSUE #114: Update dependencies to latest version
## Is your feature request related to a problem? Please describe.

The application requires NodeJS version to be `13.x.x`  because of the `node-sass 4.13.1` dependency. Considering the upgradation in the NodeJS versions over the recent times, it would be great if we can upgrade the dependencies such that they support the latest versions. This will also allow us to implement new features in the application smoothly.

## Describe the solution you'd like

- All the dependencies should be updated to their latest version, such that they are compatible with the latest stable release of NodeJS.

### PR DESCRIPTION
**Pull request by kailash360**
**Issue Number**

fixes #114 

**Describe the changes you've made**

I have made the following changes:

- Upgraded all the dependencies to their latest versions
- Added new test cases for exhaustive testing
    - For authentication
       image
    - For creating new story page
       image
    - For an individual story    
       image
    - For homepage/dashboard and it's filters
      image
    - For a user profile
      image
 

- Removed the special note to install `NodeJS 13.x.x` in the **Installation** section in `README.md` 

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

NA

**Test plan (OPTIONAL)**

- Install the dependencies using `npm i -f` command
- Run the command to run the Cypress end-to-end tests
```sh
cd cypress
npm run cypress:open
```

**Checklist**

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**

NA

<!-- Please add deployed link to here text >

### CODE REVIEW

**Comment by Abhirup-99** [comment]
General suggestion: 
- Smaller PRs are easier to review and this could have been broken into multiple PRs each adding specific tests to a set of files.
- Some files don't have a new line at the end. 

Otherwise good work!

**Comment by kailash360** [comment]
> General suggestion:
> 
> * Smaller PRs are easier to review and this could have been broken into multiple PRs each adding specific tests to a set of files.
> * Some files don't have a new line at the end.
> 
> Otherwise good work!

@Abhirup-99 
I'll keep this in mind in the next contribs. 
I have also added the newline in the Cpress test files.

**Comment by kailash360** [comment]
@Abhirup-99 
I have also fixed the `flake` error in Cypress tests for CI.

**Comment by Abhirup-99** [comment]
Hey, thanks for the PR.

General feedback: Please try claiming an issue before actually working on it. This actually fosters communications and reduces conflicts.

Please squash `minor fix` commits into 1 with a relevant commit message. Having `minor fix` doesn't look great in the commit history :).
Also, it would be much better if the entire PR can be broken at least into 2 parts(2 parts sort of making more sense right now).
One contains dependency updates and the corresponding fixes with the build fix.
One adding tests.

This would make way for a good habit and legible PRs :).

File: cypress.json [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+    "env": {
+        "testUsername": "user1",
+        "testUserEmail": "test@gmail.com",
+        "testUserPassword": "password",
+        "testStoryTitle": "This is a test story",
+        "testStoryProduct": "EOS Icons",
+        "testCategory": "Documentation"
+    },
+    "testFiles": ["auth.spec.js", "new_story.spec.js", "user_story.spec.js", "dashboard.spec.js", "user_account.spec.js"]
+}
```
Comment (Abhirup-99) [inline cypress.json:12]: no new line.

**Comment by kailash360** [comment]
> Hey, thanks for the PR.
> 
> General feedback: Please try claiming an issue before actually working on it. This actually fosters communications and reduces conflicts.
> 
> Please squash `minor fix` commits into 1 with a relevant commit message. Having `minor fix` doesn't look great in the commit history :). Also, it would be much better if the entire PR can be broken at least into 2 parts(2 parts sort of making more sense right now). One contains dependency updates and the corresponding fixes with the build fix. One adding tests.
> 
> This would make way for a good habit and legible PRs :).

Alright

File: src/components/UsersSuggestionDropdown.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import React, { useEffect, useRef, useState } from 'react'
 import axios from 'axios'
 import { trackPromise, usePromiseTracker } from 'react-promise-tracker'
-import { apiURL } from '../config.json'
+import * as config from '../config.json'
 import LoadingIndicator from '../modules/LoadingIndicator'
+const { apiURL } = config
```
Comment (Abhirup-99) [inline src/components/UsersSuggestionDropdown.js:6]: Why is it updated? You are basically writing 2 lines which were already done in 1 line.

File: src/hooks/useAuth.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import axios from 'axios'
-import { apiURL } from '../config.json'
+import * as config from '../config.json'
 import Context from '../modules/Context'
 import { useContext } from 'react'
+const { apiURL } = config
```
Comment (Abhirup-99) [inline src/hooks/useAuth.js:5]: same here.

File: src/components/UsersSuggestionDropdown.js (hunk 2 above)
Comment (kailash360) [reply src/components/UsersSuggestionDropdown.js:6]: It was returning a webpack-based error and creating issue in tests, so I had to replace it.

File: src/services/api.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import axios from 'axios'
-import { apiURL, APP_ENV } from '../config.json'
+import * as CONFIG from '../config.json'
+const { apiURL, APP_ENV } = CONFIG
```
Comment (Abhirup-99) [inline src/services/api.js:3]: same here.

File: src/components/UsersSuggestionDropdown.js (hunk 2 above)
Comment (Abhirup-99) [reply src/components/UsersSuggestionDropdown.js:6]: what was the error?

Comment (kailash360) [reply src/components/UsersSuggestionDropdown.js:6]: > Member



> what was the error? The path should be to fix the error, not break something which was done in the correct way.

image

Comment (kailash360) [reply src/components/UsersSuggestionDropdown.js:6]: @Abhirup-99 This was the error

Comment (Abhirup-99) [reply src/components/UsersSuggestionDropdown.js:6]: Didn't realise it was a JSON. Ignore comments.

File: cypress.json (hunk 1 above)
Comment (kailash360) [reply cypress.json]: Alright, I'll add one there

**Comment by kailash360** [comment]
@Abhirup-99 
Kindly close this PR, I have created PR #117 for upgrading the dependencies and fixing the breaking errors.

**Pull request closed without merging**